src/lib/useDomainSearch.tsx

```diff
@@ -20,31 +20,39 @@ export function useDomainSearch() {
 
 	useAsyncEffect(async () => {
 		const id = getDomainId(pattern);
-		try {
-			const rawDomain = await sdk.getDomainById(id);
-			const exactDomain = rawDomain as any;
-			exactDomain.metadata = exactDomain.metadataUri;
-			exactDomain.minter = { id: exactDomain.minter };
-			exactDomain.owner = { id: exactDomain.owner };
-			exactDomain.parent = { id: exactDomain.parentId };
-			exactDomain.lockedBy = { id: exactDomain.lockedBy };
+		if (id) {
+			try {
+				const rawDomain = await sdk.getDomainById(id);
```

Comment:
Maybe not necessary for this iteration, but I think we could speed this up by removing `sdk.getDomainById` for exact match.

Something like:
```
get domains by name
filter domains for exact match of search query
```

This way there would be one API call per search query rather than two.

Reply:
@colbr  Yeah makes sense. We can do that. I also think we need to put this behind a debounce to avoid frequent calls as user types ahead.